presto-main/src/main/java/io/prestosql/operator/project/MergePages.java

```diff
@@ -127,7 +129,7 @@ private MergePagesTransformation(Iterable<? extends Type> types, long minPageSiz
                 return ofResult(flush(), false);
             }
 
-            if (inputPage.getPositionCount() >= minRowCount || inputPage.getSizeInBytes() >= minPageSizeInBytes) {
+            if (inputPage.getPositionCount() >= minRowCount || containsNotLoadedLazyBlock(inputPage) || inputPage.getSizeInBytes() >= minPageSizeInBytes) {
```

Comment:
> For this operator, I think you can still handle the < minRowCount case with lazy materialization

Added TODO

Reply:
> For this operator, I think you can still handle the < minRowCount case with lazy materialization

Such merged lazy block would have more rows. If any row is touched, then all subblocks would be materialized. I think we could solve that by extending lazy page to support projected load (this could also improve performance of lazy ORC decoding for large chunks with few rows selected).